stablehlo/dialect/StablehloOps.td

```diff
@@ -2471,14 +2531,19 @@ def StableHLO_TransposeOp: StableHLO_ShapedInterfaceOp<"transpose",
 
     Example:
     ```mlir
-    %0 = "stablehlo.transpose"(%arg0) {permutation = dense<[2, 1, 0]> : tensor<3xi64>} : (tensor<2x3x2xi32>) -> tensor<2x3x2xi32>
+    %0 = stablehlo.transpose %arg0, perm = [2, 1, 0] : (tensor<1x2x3xi32>) -> tensor<3x2x1xi32>
```

Comment:
`dims` to be consistent with most of the other ops in this PR, as well as [HLO Text](https://github.com/tensorflow/tensorflow/blob/60c6715d7513e7bcbee6e670b2d5eb0432a831c3/tensorflow/compiler/xla/translate/hlo_to_mhlo/tests/import.hlotxt#L1389)?